test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreLoggingTests.cs

```diff
@@ -139,6 +140,101 @@ public async Task RestoreLogging_VerifyNU1608MessageAsync()
+                            <version>1.0.0</version>
+                            <title />
+                        </metadata>
+                        </package>")
+                };            
+
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(pathContext.PackageSource, packageX);
+                projectA.AddPackageToAllFrameworks(packageX);                
+
+                solution.Projects.Add(projectA);
+                solution.Create(pathContext.SolutionRoot);
+                
+                File.Delete(Path.Combine(pathContext.PackageSource, packageX.Id, packageX.Version, packageX.Id + NuGetConstants.ManifestExtension));
+
+                // Act                
+                var r = Util.RestoreSolution(pathContext, expectedExitCode: 1);
+
+                // Assert
+                r.Success.Should().BeFalse();                
+                r.AllOutput.Should().Contain("The package doesn't contain .nuspec file: " + Path.Combine(pathContext.PackageSource, packageX.Id, packageX.Version));
+            }
+        }
+
+        [Fact]
+        public async Task RestoreLogging_VerifyNU5036ErrorMessageWithDependencyAsync()
+        {
+            // Arrange
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                // Set up solution, project, and packages
+                var solution = new SimpleTestSolutionContext(pathContext.SolutionRoot);
+
+                var netcoreapp1 = FrameworkConstants.CommonFrameworks.NetCoreApp10;
+
+                var projectA = SimpleTestProjectContext.CreateNETCore(
+                    "a",
+                    pathContext.SolutionRoot,
+                    netcoreapp1);
+
+                var packageX = new SimpleTestPackageContext("x", "1.0.0")
+                {
+                    Nuspec = XDocument.Parse($@"<?xml version=""1.0"" encoding=""utf-8""?>
+                        <package>
+                        <metadata>
+                            <id>x</id>
+                            <version>1.0.0</version>
+                            <title />
+                            <dependencies>
+                                <group>
+                                    <dependency id=""z"" version=""[1.0.0]"" />
+                                </group>
+                            </dependencies>
+                        </metadata>
+                        </package>")
+                };
+
+                var packageZ1 = new SimpleTestPackageContext("z", "1.0.0");
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(pathContext.PackageSource, packageX, packageZ1);
+                projectA.AddPackageToAllFrameworks(packageX);
+
+                solution.Projects.Add(projectA);
+                solution.Create(pathContext.SolutionRoot);
+
+                File.Delete(Path.Combine(pathContext.PackageSource, packageZ1.Id, packageZ1.Version, packageZ1.Id + NuGetConstants.ManifestExtension));
```

Comment:
This test is doing the exact same thing as the previous one.
It doesn't introduce a new relevant scenario.

Reply:
The first test is only checking the restore of the root package. The second unit test is to verify behavior when .nuspec file is deleted from the dependency package folder.
For e.g, package x is dependent on z.1.0.0. When .nuspec file is deleted from z.1.0.0 folder and package x is restored then the tooling should throw this NU5036 error message with required info.

Reply:
I guess the key is relevant. 

It doesn't matter which package's nuspec was deleted as long as the package is within the resolved graph.